frontend/src/lib/services/FetchProducts.ts

```diff
@@ -0,0 +1,109 @@
+import Product from 'models/Product';
+import Version from 'models/Version';
+import { doGetRequest } from 'lib/rest/Rest';
+
+interface ResponseDataElement {
+  _id: string;
+  favourite: boolean;
+  builds: string[];
+  pinned: string[];
+  coordinates: {
+    product: string;
+    major: string;
+    minor: string;
+    servicePack: string;
+  };
+}
+type ResponseData = ResponseDataElement[];
+
+const isExpectedResponse = (responseData: unknown): responseData is ResponseData => {
+  if (!Array.isArray(responseData)) {
+    return false;
+  }
+  if (responseData.some(element => element._id === undefined)) {
+    return false;
+  }
+  if (responseData.some(element => element.favourite === undefined)) {
+    return false;
+  }
+  if (
+    responseData.some(
+      element => element.builds === undefined || !Array.isArray(element.builds) || element.builds.some(build => typeof build !== 'string')
```

Comment:
Probs don't need the undefined check here the array one will do it. 

I am thinking these things are going to be used over and over while deserialising responses. Might be work extracting them to some util functions at some point.

Something like 
```
getArrayOfStringsChecker = (attr: string) => (element: any): boolean => !Array.isArray(element[attr]) || element[attr].some(thing => typeof thing !== 'string')
```

Reply:
Ah yep I'll remove the undefined check. Yeah I'm not 100% happy with this atm, the main two issues are the boilerplate stuff and the fact that if it fails you don't know which part failed. I'll try to come up with something for these in the next one.